GridList/GridList.jsx

```diff
@@ -0,0 +1,178 @@
+import { h } from "preact";
+import MaterialComponent from "../MaterialComponent";
+
+/**
+ * @prop cellHeight	{number} - Number of px for one cell height. You can set 'auto' if you want to let the children determine the height.
+ * @prop cols {number} - Number of columns.
+ * @prop padding {1|4} - Number of px for the padding/spacing between items.
+ * @prop style {object} - Override the inline styles of the root element
+ */
+class GridList extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-list";
+    this._propsDict = {
+      cols: "cols",
+      padding: "padding",
+      cellHeight: "cellHeight",
+      secondaryActionIconAlign: "secondaryActionIconAlign",
+      aspectRatio: "aspectRatio",
+      twoLineCaptions: "twoLineCaptions"
+    };
+  }
+  createClassName(props) {
+    const baseClass = "mdc-grid-list";
+    const classes = [];
+
+    if (
+      props[this._propsDict.padding] &&
+      props[this._propsDict.padding] === 1
+    ) {
+      classes.push("mdc-grid-list--tile-gutter-1");
+    }
+
+    if (props[this._propsDict.secondaryActionIconAlign]) {
+      classes.push(
+        "mdc-grid-list--with-icon-align-" +
+          props[this._propsDict.secondaryActionIconAlign]
+      );
+    }
+
+    if (props[this._propsDict.aspectRatio]) {
+      classes.push(
+        "mdc-grid-list--tile-aspect-" + props[this._propsDict.aspectRatio]
+      );
+    }
+
+    if (props[this._propsDict.twoLineCaptions]) {
+      classes.push("mdc-grid-list--twoline-caption");
+    }
+
+    return classes.join(" ");
+  }
+  materialDom(props) {
+    const cols = props.cols || 2;
+    const tileWidthPct = Math.floor(100 / cols) + "%";
+
+    return (
+      <span>
+        <style
+          dangerouslySetInnerHTML={{
```

Comment:
I don't think that this belongs in the components. Wouldn't it be far cleaner to just add this to an external CSS?

Reply:
Yes, I struggled with this one. I'm not a fan of this, but I'm not sure how we could support the `cols` prop that is supported http://www.material-ui.com/#/components/grid-list without something like this.

I do believe that it's a nicer interface to say, "give me a gridlist that has X columns" rather than having to add your own style rule, particularly if you are generating this programmatically.

However, if there is a strong desire to avoid this sort of embedded styles, I can remove it.

Reply:
@filoozom / @prateekbh - what are your thoughts about embedding the style to make it more declarative?

Reply:
Well, it would work with SSR so there's that, but I'm really not a fan of it.